### PR TITLE
Adding design doc for network policy support

### LINKED ISSUE #257: Design Spec of network policy in Mizar
This is the Google Doc link to our network policy design & discussion:

https://docs.google.com/document/d/1zgs_iAVwU3DN72FNGU6KTGFmmM8xTiRFWjXJXz6q-R0/edit?usp=sharing

Once the design is finished, the doc will be converted to a PR.

### PR DESCRIPTION
**Pull request by XiaoningDing**
Based on the design draft & discussion records made by Hongwei, Cathy and Hong:

https://docs.google.com/document/d/1zgs_iAVwU3DN72FNGU6KTGFmmM8xTiRFWjXJXz6q-R0/edit#heading=h.jsvmos9xw7p7

It's the first version of the design doc. Some sections are marked as TBD.  Once we figure out the details we will update them.

### CODE REVIEW

File: docs/design/dp_network_policy.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+   dip_prim_rule = Dest-IP-range-primary-table[packet.VNI, packet.source-ip, packet.dest-ip];
+   dip_supp_rules = Dest-IP-range-supplementary-table[packet.VNI, packet.source-ip, packet.dest-ip];
+   dip_except_rules = dest-ip-range-excpet-table[packet.VNI, packet.source-ip, packet.dest-ip];
+   pp_l4_rules = Proto-port[packet.VNI, packet.source-ip, packet.dest-port, packet.protocol];
+   pp_l3_rules = Proto-port[packet.VNI, packet.source-ip, 0, 0 ];
+
+
+   //perform the bit operation to the get final result, to see if there is any policy allows the packet
+   result =  (dip_prim_rule | (dip_supp_rules & ~dip_except_rules)) & ( pp_l4_rules  | pp_l4_rules )
```
Comment (h-w-chen) [inline docs/design/dp_network_policy.md]: pp_l4_rules | pp_l**3**_rules

File: docs/design/dp_network_policy.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+### Mizar Daemon
+
+* New RPC interfaces are implemented to receive the network policy CRUD messages from Mizar master, and then updates the corresponding BPF maps in the kernel space. 
+* Delete invalid entries in connection tracking based on the updated policies.
+
+
+### Transit XDP
+
+* Call the connection tracking module to:
```
Comment (h-w-chen) [inline docs/design/dp_network_policy.md]: suggest to re-format this block; it seems the invokation of connection track nodule is the only controlling factor by current formating.

**Review by h-w-chen — APPROVED** [review]
I signed off at areas of policy check, transit and agent.

**Review by clu2xlu — APPROVED** [review]

File: docs/design/dp_network_policy.md (hunk 1 above)
Comment (XiaoningDing) [reply docs/design/dp_network_policy.md]: Thanks. Fixed.

File: docs/design/dp_network_policy.md (hunk 2 above)
Comment (XiaoningDing) [reply docs/design/dp_network_policy.md]: Updated.

**Pull request merged**